doc/nrf/app_dev/device_guides/hpf/index.rst

```diff
@@ -0,0 +1,15 @@
+.. _coprocessors_index:
+
+Developing with coprocesors
+###########################
+
+Thie following pages provide a detailed guide for developers aiming to utilize coprocessors effectively within the nRF54L and nRF54H platforms.
+The RISC-V coprocessor integrated into these platforms facilitates the development of custom, software-defined peripherals, enhancing the functionality and performance of the main processor.
```

Comment:
This section should not focus on HPF, but explain that you can use VPR with both:
- multithreaded zephyr (and that page should be taken from https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/app_dev/device_guides/nrf54l/vpr_flpr.html)
- HPF (and the whole structure added here should then be under that)